prog/main.go

```
@@ -134,6 +138,62 @@ type appFlags struct {
 	consulInf       string
 }
 
+type containerLabelFiltersPositive []app.APITopologyOption
+
+func (filters *containerLabelFiltersPositive) String() string {
+	return fmt.Sprint(*filters)
+}
+
+func (filters *containerLabelFiltersPositive) Set(value string) error {
+	*filters = append(*filters, makeAPITopologyOption(true, value, len(*filters)))
+	return nil
+}
+
+type containerLabelFiltersNegative []app.APITopologyOption
+
+func (filters *containerLabelFiltersNegative) String() string {
+	return fmt.Sprint(*filters)
+}
+
+func (filters *containerLabelFiltersNegative) Set(value string) error {
+	*filters = append(*filters, makeAPITopologyOption(false, value, len(*filters)))
+	return nil
+}
+
+func makeAPITopologyOption(positive bool, value string, filterNumber int) app.APITopologyOption {
+	u := fmt.Sprintf("%q", value)
+	u = u[1 : len(u)-1]
+
+	indices := colonFinder.FindStringIndex(u)
+	if len(indices) == 0 {
+		log.Infof("No unescaped colon found. This is needed to separate the title from the label")
```

Instead of logging, please propagate errors to the flag parser (until `Set()`).